projects/giflib/egif_fuzz_common.cc

```diff
@@ -1,26 +1,36 @@
 #include "egif_fuzz_common.h"
-#define GIF_IMAGE_WIDTH  100
+#define GIF_IMAGE_WIDTH 100
 // This is rgb byte stream length per horizontal line = GIF_IMAGE_WIDTH * 3
-#define GIF_IMAGE_LINE   300
+#define GIF_IMAGE_LINE 300
 
 extern "C" void PrintGifError(int ErrorCode);
 
-int stub_output_writer(GifFileType* gifFileType, const uint8_t* buf, int len)
+int stub_output_writer(GifFileType *gifFileType, const uint8_t *buf, int len)
 {
-	struct gifUserData* gud = (struct gifUserData*) gifFileType->UserData;
+	struct gifUserData *gud = (struct gifUserData *)gifFileType->UserData;
 
 	if (gud == NULL || gud->gifData == NULL || len == 0)
 		return 0;
-
-	memcpy(gud->gifData, buf, len);
-	gud->gifData += len;
+	if (gud->allocatedSize < (gud->gifLen + len))
+	{
+		// Reallocate gifFileType
+		int newSize = gud->allocatedSize * 2;
+		uint8_t *oldGud = gud->gifData;
+		gud->gifData = (uint8_t *)realloc(oldGud, newSize);
+		if (gud->gifData == NULL)
+		{
+			gud->gifData = oldGud;
+			return 0;
```

Comment:
Should we add a debug print here under the FUZZING_MODE_UNSAFE_FOR_PRODUCTION macro so we know how often, if at all, this happens?

Reply:
Yup - my intuition here is that this should only happen when the target is OOM, in which case the fuzzer should complain anyway, correct?

Reply:
Good point. To be safe, we could add `assert(0)` here so we know if some input gets us here before OOM.

Reply:
Will do:)

Reply:
Fixed